### PR TITLE
+ stream add runHead and runHeadOption to Source,useful when write te…

### PR DESCRIPTION
**Pull request by He-Pin**
this is useful when writing some test code, especially in Java.

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/javadsl/Source.scala [hunk 1]
```diff
@@ -940,6 +939,27 @@ final class Source[+Out, +Mat](delegate: scaladsl.Source[Out, Mat]) extends Grap
...
+  /**
+   * Shortcut for running this `Source` to retrieve the first value received.
+   * If the stream completes before signaling at least a single element, the CompletionStage will be failed with a [[NoSuchElementException]].
+   * If the stream signals an error before signaling at least a single element, the CompletionStage will be failed with the streams exception.
+   *
+   * See also [[runHeadOption]].
+   */
+  def runHead(implicit materializer: Materializer): CompletionStage[Out @uncheckedVariance] = {
+    runWith(Sink.head(), materializer)
+  }
```
Comment (hepin1989) [inline akka-stream/src/main/scala/akka/stream/javadsl/Source.scala]: will remove this later

File: akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala [hunk 2]
```diff
@@ -138,6 +138,26 @@ final class Source[+Out, +Mat](
...
   def runForeach(f: Out ⇒ Unit)(implicit materializer: Materializer): Future[Done] = runWith(Sink.foreach(f))
 
+  /**
+   * Shortcut for running this `Source` to retrieve the first value received.
+   * If the stream completes before signaling at least a single element, the Future will be failed with a [[NoSuchElementException]].
+   * If the stream signals an error errors before signaling at least a single element, the Future will be failed with the streams exception.
+   *
+   * See also [[runHeadOption]].
+   */
+  def runHead(implicit materializer: Materializer): Future[Out] =
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala]: why does `head` deserve its own `run`? Why isn't `runWith(Sink.head)` convenient enough?

Comment (hepin1989) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala]: like `runWith(Sink.foreach(f))`, I think `runWith(Sink.head)` is so common so, with this method, I could write all my code in the domain of `Source[T]` not both.

Comment (patriknw) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala]: Common? Perhaps in tests but I can’t imagine that it is very interersting for real streams.

Comment (hepin1989) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala]: When I am using this, I see the *Stream* as a *Stream* of element `0 to 1`, for this kind of, it's common :)

Comment (ktoso) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala]: That does not really address the question of it being really common and what we should show people. People *should* learn about Sinks and Sources early on. It's the key feature and mindset-thing of akka streams

**Comment by patriknw** [comment]
Closing this since we are not convinced that it has enough value to motivate the API growth.

**Pull request closed without merging**